modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelectionStrategy.scala

```diff
@@ -0,0 +1,53 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.kernel.Async
+import cats.effect.std.Random
+import cats.syntax.all._
+
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.schema.peer.Peer.toP2PContext
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.peerSelection.{NoPeers, PeerSelectionStrategy}
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+
+class MajorityPeerSelectionStrategy[F[_]: Async: KryoSerializer: Random](
+  val storage: ClusterStorage[F],
+  val snapshotClient: L0GlobalSnapshotClient[F]
+) extends PeerSelectionStrategy[F] {
+  def select: F[Peer] = for {
+    readyPeers <- storage.getResponsivePeers
+      .map(_.filter(_.state === Ready))
+    peersToQuery <- getPeerSublist(readyPeers).flatMap { peers =>
+      Async[F].fromOption(peers.toNel, NoPeers)
```

Comment:
@marcinwadon Couldn't an argument be made that `Async` is the right typeclass to use because we'd then have to specify the error type parameter: doesn't that have to match the one defined in `Async[F] <- Sync[F] <- MonadCancel[F, Throwable] <- ...` ?

Reply:
but then you don't use the least power. You can use Async everywhere but then tagless algebra makes no sense

Reply:
I agree, but that's not what I'm saying. I'm asking whether or not there is a dependency on us using the same error type as `Async` is using for the ApplicativeError. If there is, then we should just go through the class we're using as the constraint. If there is not, then we can change it to the defining class.

Reply:
Could you show an example? I don't fully understand what you mean

Reply:
What I want to understand is whether we have a dependency on the defined constraint. In this code it's `Async`.

For example, given Sync is defined like this,
```
trait Sync[F[_]] extends MonadCancel[F, Throwable] with Clock[F] with Unique[F] with Defer[F]
```
here are two ways to call `fromOption`
```
Sync[F].fromOption(...)  // uses error type parameter defined in Sync
ApplicativeError[F, E].fromOption(...)
```
What should we set `E` to?  If `Throwable`, then why? Is it because our function requires evidence of `Sync[F]`? If so then we should just use `Sync` otherwise we're assuming we have knowledge of the implementation.

If we can set `E` to anything else and not change the correctness or get compiler error, then `ApplicativeError` is fine.

Reply:
In other words, does the `E` we use in `ApplicativeError` have to match the type parameter used in the typeclass we're constrained against, which in this case is `Async`.

Reply:
we use `MonadThrow` in many places which is an alias that sets `E` to `Throwable`.
I'm good also to use `MonadThrow[F].fromOption(..)` but effectively using `MonadThrow`/`Sync`/`Async` ends up with `ApplicativeError[F, Throwable]`.
when answering to `why` - `Throwable` is the most generic type for errors. but if you want, I think you can pass more specified type there

Reply:
Changed to using `MonadThrow`.